code/to_bsdd.py

```diff
@@ -197,39 +230,56 @@ def get_parent_of_pt(enum_type):
                         enum_types_by_name += [c for c in xmi_doc.xmi.by_tag_and_type["packagedElement"]["uml:Enumeration"] if c.name == type_name]
                         type_values = [x.name for x in enum_types_by_name[0]/"ownedLiteral"]
-                    elif ty == "PropertySingleValue":
+                        kind_name = 'Single'
+                    else:
                         type_name = list(ty_arg.values())[0]
                         pe_types = [c for c in xmi_doc.xmi.by_tag_and_type["packagedElement"]["uml:Class"] if c.name == type_name]
                         pe_types += [c for c in xmi_doc.xmi.by_tag_and_type["packagedElement"]["uml:DataType"] if c.name == type_name]                    
                         pe_type = pe_types[0]
                         root_generalization = generalization(pe_type)
                         type_name = root_generalization.name.lower()
                         type_values = None
-                    else:
-                        print("Warning: %s.%s of type %s <%s> not mapped" % (item.name, nm, ty, ",".join(map(lambda kv: "=".join(kv), ty_arg.items()))))
-                        continue
+                        if ty == "PropertySingleValue":
+                            kind_name = 'Single'
+                        elif ty == "PropertyBoundedValue":
+                            di["Psets"][item.name]["Properties"][a.name]["Description"] = "This property in IFC stores a bounded value, meaning it has a maximum of two (numeric or descriptive) values assigned, the first value specifying the upper bound and the second value specifying the lower bound. Read the IFC documentation for more information."
+                            kind_name = 'Range'
+                        elif ty == "PropertyReferenceValue":
+                            di["Psets"][item.name]["Properties"][a.name]["Description"] = "This property in IFC takes as its value a reference to one of: IfcAddress, IfcAppliedValue, IfcExternalReference, IfcMaterialDefinition, IfcOrganization, IfcPerson, IfcPersonAndOrganization, IfcTable, IfcTimeSeries. Read the IFC documentation for more information."
+                            kind_name = 'Complex'
+                        elif ty == "PropertyListValue":
+                            di["Psets"][item.name]["Properties"][a.name]["Description"] = "This property in IFC takes list as its value. Read the IFC documentation for more information."
+                            kind_name = 'List'
+                        elif ty == "PropertyTableValue":
+                            di["Psets"][item.name]["Properties"][a.name]["Description"] = "This property in IFC takes a table as its value. That table has two columns (two lists), one for defining and other for defined values. Read the IFC documentation for more information."
+                            kind_name = 'Complex'
```

Comment:
```suggestion
                            kind_name = 'Table'
```

?

Reply:
those are bSDD kinds, and they can be Single, List, Complex, ComplexList, Range.